_projects/2024/100508856/100508856.Rmd

```diff
@@ -0,0 +1,333 @@
+
+
+## First two graphs
+
+The code for this visual representation can be divided into four parts:
+
+1. Modify column names after reading the corresponding data table
+2. Set the X-axis as the year, the Y-axis as the working time, and the colors are differentiated by region
+3. Modify the title to "Annual Working Hours per Country from 1950 to 2017"
+4. Modify the x-axis of the plot
+
+
+```{r}
+
+
+# Figure 1
+library(ggplot2)
+library(dplyr)
+library(ggrepel)
+library(tidyr)
+devtools::install_github("malcolmbarrett/ggokabeito")
+library(ggokabeito)
+library(paletteer)
+working_hours<-read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual working hours/annual-working-hours-per-worker.csv")
+colnames(working_hours)<-c("Region","Code","Year","working_hours")
+p=ggplot(working_hours, aes(x = Year, y = working_hours, group = Region, color = Region)) +
+   geom_line() +
+    theme_minimal() +
+   labs(title = "Annual Working Hours per Country from 1950 to 2017",
+                  x = "Year",
+                  y = "working Hours",
+                  color = "Region") +
+     scale_x_continuous(breaks = seq(1950, 2017, by = 10)) 
+ 
+p
+```
+
+```{r}
+#Figure 2
+
+data <- read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual working hours/annual-working-hours-per-worker.csv")
+colnames(data)[4] <- "time"
+data1 <- dplyr::filter(data, grepl('Brazil|China|France|Germany|Hong Kong|India|
+                                   Singapore|Taiwan|United Kingdom|United States', Entity))
+
+p<-ggplot(data1,aes(Year,time,group=Entity,color=Entity))+
+  geom_line(position = position_dodge(0.1),cex=1)+
+  scale_x_continuous(breaks = seq(1870, 2020, 30))+
+  theme_bw()+
+  labs(x="Year",
+       y="Hours",
+       color="Country/Region")+
+  scale_color_manual(values=paletteer_d("ggthemes::Classic_20", n=9))+
+  theme(panel.grid.major.x = element_blank(),
+        panel.grid.minor.x = element_blank(),
+        panel.grid.major.y = element_line(linetype=2),
+        panel.grid.minor.y = element_blank(),
+        axis.text=element_text(size = 13), 
+        axis.title =element_text(size = 15),
+        plot.margin=unit(c(3,0,3,0),'cm'),
+        panel.border = element_blank(),
+        axis.line.x= element_line(colour = "black"))
+p
+
+
+```
+
+
+## Maps
+
+## steps
+
+1. Select the data for the corresponding year and select the required column name (entity and time).
+2. Change the column name and merge data
+3. Modify the Settings of the map: set the title, control the title text size, scale size, map proportion, text color, etc
+
+
+```{r}
+#Figure 3 maps
+
+library(tmap)
+data("World")
+
+##The map in 1900
+data_1900 <- data[data$Year==1900,] %>% select("Entity","time")
+colnames(data_1900) <- c("sovereignt","Hours")
+world_1900 <- left_join(World,data_1900,by="sovereignt")
+tm_shape(world_1900) +
```

Comment:
The course is about ggplot2, and we have studied how to plot maps with ggplot2. ;-) Please, use ggplot2 for these maps.